src/main/java/graphql/annotations/GraphQLAnnotations.java

```diff
@@ -588,12 +588,21 @@ private GraphQLOutputType internalGetGraphQLConnection(AccessibleObject field, G
         assert wrappedType instanceof GraphQLObjectType;
         String connectionName = field.getAnnotation(GraphQLConnection.class).name();
         connectionName = connectionName.isEmpty() ? wrappedType.getName() : connectionName;
-        GraphQLObjectType edgeType = relay.edgeType(connectionName, (GraphQLOutputType) wrappedType, null, Collections.<GraphQLFieldDefinition>emptyList());
-        type = relay.connectionType(connectionName, edgeType, Collections.emptyList());
+        GraphQLObjectType edgeType = ensureTypeUniqueness(relay.edgeType(connectionName, (GraphQLOutputType) wrappedType, null, Collections.<GraphQLFieldDefinition>emptyList()));
```

Comment:
ensure suggests that the function is boolean. Please rename it to something better
(like getUniqueType, or getActualType)